src/main/java/com/pinterest/secor/io/impl/FlexibleDelimitedFileReaderWriterFactory.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+
+
+  protected class FlexibleDelimitedFileReader implements FileReader {
+    private final BufferedInputStream mReader;
+    private long mOffset;
+    private Decompressor mDecompressor = null;
+    private byte DELIMITER = delimterReader();
+
+    public FlexibleDelimitedFileReader(LogFilePath path, CompressionCodec codec) throws IOException {
+      Path fsPath = new Path(path.getLogFilePath());
+      FileSystem fs = FileUtil.getFileSystem(path.getLogFilePath());
+      InputStream inputStream = fs.open(fsPath);
+      this.mReader = (codec == null) ? new BufferedInputStream(inputStream)
+      : new BufferedInputStream(
+      codec.createInputStream(inputStream,
+      mDecompressor = CodecPool.getDecompressor(codec)));
+      this.mOffset = path.getOffset();
+    }
+
+    public byte delimterReader() {
+      byte delimiter = '\n';
+      try {
+        String readerDelimiter = SecorConfig.load().getFileReaderDelimiter();
+        if (!readerDelimiter.isEmpty()){
+          delimiter = (byte)readerDelimiter.charAt(0);
+        }
+      }catch(ConfigurationException e){}
+        return delimiter;
+      }
+
+      @Override
+      public KeyValue next() throws IOException {
+        ByteArrayOutputStream messageBuffer = new ByteArrayOutputStream();
+        int nextByte;
+        while ((nextByte = mReader.read()) != DELIMITER) {
+          if (nextByte == -1) { // end of stream?
+            if (messageBuffer.size() == 0) { // if no byte read
+              return null;
+            } else { // if bytes followed by end of stream: framing error
+              throw new EOFException(
+              "Non-empty message without delimiter");
+            }
+          }
+          messageBuffer.write(nextByte);
+        }
+        return new KeyValue(this.mOffset++, messageBuffer.toByteArray());
+      }
+
+      @Override
+      public void close() throws IOException {
+        this.mReader.close();
+        CodecPool.returnDecompressor(mDecompressor);
+        mDecompressor = null;
+      }
+    }
+
+    protected class FlexibleDelimitedFileWriter implements FileWriter {
+      private final CountingOutputStream mCountingStream;
+      private final BufferedOutputStream mWriter;
+      private Compressor mCompressor = null;
+      private byte DELIMITER = delimterWriter();
+
+      public FlexibleDelimitedFileWriter(LogFilePath path, CompressionCodec codec) throws IOException {
+        Path fsPath = new Path(path.getLogFilePath());
+        FileSystem fs = FileUtil.getFileSystem(path.getLogFilePath());
+        this.mCountingStream = new CountingOutputStream(fs.create(fsPath));
+        this.mWriter = (codec == null) ? new BufferedOutputStream(
+        this.mCountingStream) : new BufferedOutputStream(
+        codec.createOutputStream(this.mCountingStream,
+        mCompressor = CodecPool.getCompressor(codec)));
+      }
+
+      public byte delimterWriter() {
+        byte delimiter = '\\';
```

Comment:
You should prevent other developers misuse this parameter if you are only going to take char(0), the check is needed to prevent those misuses.